### ISSUE TITLE
Go-to-def/find-all-refs/rename doesn't work for class declared after type alias

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/pylance-release#5304]
Raw code:

```python
type LinkedList = Node | None

class Node:
    def __init__(self, value: object, next: LinkedList):
        self.value = value
        self.next = next
```

```python
type LinkedList = Node$1$ | None

class Node$2$:
    def __init__(self, value: object, next: LinkedList):
        self.value = value
        self.next = next
```

Go-to-definition at `$1$`, or rename/find-all-refs at either `$1$` or `$2$`.

* neither go-to-def nor rename work at all on `$1$`
* rename/find-all-refs on `$2$` misses the usage of `Node` at the type alias.

The same issue is present if using a "legacy" type alias.

```python
from typing import TypeAlias

LinkedList: TypeAlias = "Node | None"

class Node:
    def __init__(self, value: object, next: LinkedList):
        self.value = value
        self.next = next
```

**Comment by erictraut** [comment]
Thanks for the bug report. This was a bug in the core type checker, so I've fixed it in the pyright code base.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.12.104, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202312104-22-January-2024-PreRelease)
